### PR TITLE
[`Style`]: use standard style of `ReadOnlyStore.TryGet`

### PR DESCRIPTION
**Pull request by nan01ab**
# Description

The previous `ReadOnlyStore.TryGet` returns `null` when key doesn't exist, but this style is not a common style in C#,
Different styles make it weird.

So this PR makes `byte[] ReadOnlyStore.TryGet(byte[] key)` obsolete, and uses `bool ReadOnlyStore.TryGet(byte[] key, out byte[] value`) instead(added in #3533)

Fixes # (issue)

## Type of change

- [ ] Optimization (the change is only an optimization)
- [x] Style (the change is only a code style for better maintenance or standard purpose)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by Jim8y — DISMISSED** [review]

**Review by shargon — APPROVED** [review]

File: src/Neo/Persistence/IReadOnlyStore.cs [hunk 1]
```diff
@@ -32,7 +32,7 @@ public interface IReadOnlyStore
         /// </summary>
         /// <param name="key">The key of the entry.</param>
         /// <returns>The data of the entry. Or <see langword="null"/> if it doesn't exist.</returns>
-        /// <seealso cref="TryGet(byte[], out byte[])"/>. Obsolete it later for avoiding complier warning.
+        [Obsolete("use TryGet(byte[] key, out byte[] value) instead.")]
```
Comment (cschuchardt88) [inline src/Neo/Persistence/IReadOnlyStore.cs:35]: can you add for `TryGet` on the `out` parameter; add attribute `NotNullWhen(true)`

Comment (cschuchardt88) [reply src/Neo/Persistence/IReadOnlyStore.cs:35]: Looks like it already committed.

Comment (cschuchardt88) [reply src/Neo/Persistence/IReadOnlyStore.cs:35]: I was talking about the C# analyzers.

**Review by cschuchardt88 — APPROVED** [review]

File: src/Neo/Persistence/IReadOnlyStore.cs (hunk 1 above)
Comment (shargon) [inline src/Neo/Persistence/IReadOnlyStore.cs:35]:
```suggestion
        [Obsolete("use TryGet(byte[] key, [NotNullWhen(true)] out byte[]? value) instead.")]
```

**Review by Jim8y — APPROVED** [review]

**Pull request merged**